src/main/webapp/mapsScript.js

```diff
@@ -90,6 +93,93 @@ function showDirections(locations) {
       window.alert('Directions request failed due to ' + status);
     }
   });
+
+  // Add markers to map.
+  addPoiMarker(origin, true, 0);  
+  for (let i = 0; i < waypts.length; i++) {
+    let index = i + 1;
+    addPoiMarker(waypts[i].location, false, index);
+  }
```

Comment:
Note: I know that I am looping through the waypts twice (once to parse and once to add markers), however I have to add these markers after the directions have been calculated so I can query for their place IDs in the vicinity of the map center (once the directionsRenderer has recentered the map).

Reply:
I don't think this loop necessarily happens after the directionsRenderer is called. The directionsRenderer is inside of the callback function which could be executed much later.

It seems like it might be working anyways, in which case you may be able to condense the two loops into one.

Reply:
I played around with combining the loops, but the text search was searching from a center of (0,0) rather than the new center of the map. Added an inline comment explaining why this second loop is needed.